spec/grid_spec.rb

```diff
@@ -0,0 +1,20 @@
+require "spec_helper"
+require_relative "../lib/grid.rb"
+
+RSpec.describe Grid do
+  describe "#coordinate_valid?" do
+    it "says a valid coordinate is in the coordinate values" do
+      coordinate = "A3"
+      grid = Grid.new
+
+      expect(grid.coordinate_valid?(coordinate)).to eq true
+    end
+
+    it "says an invalid coordinate is not in the coordinate values" do
+      coordinate = "Z6"
+      grid = Grid.new
+
+      expect(grid.coordinate_valid?(coordinate)).to eq false
+    end
+  end
+end
```

Comment:
Is this too simple, is that the idea, test everything? It definitely is useful!!

Reply:
These are two scenarios, but if you categorize them they are:
1. Is valid in the grid somewhere
2. Is invalid if not on the grid

Whenever you have a boolean method like this, figuring out what each of the boundaries are and testing each of the boundaries is important for regression and also for documentation. These are possible boundaries, but you should figure out what is the most efficient way to test any possible boundary by testing only the ones the ones that are sufficient.

* The row is too low to be on the grid
* The row is too high to be on the grid
* The column is too low to be on the grid
* The column is to high to be on the grid
* The input is too many characters
* The input is too few characters
* The input doesn't have the right format
* What happens in the first character is lower case

You don't want to exhaustively test each coordinate on the grid, either. Just the boundaries.